File: arrays/3sum.py
class Solution:
    def threeSum(self, nums: List[int]) -> List[List[int]]:
        result = []
        n = len(nums)
        for i in range(n):
            for j in range(n):
                if i == j:
                    continue
                for k in range(n):
                    if i == k or j == k:
                        continue
                    triplet = [nums[i], nums[j], nums[k]]
                    if sum(sorted(triplet)) == 0:
                        result.append(tuple(sorted(triplet)))
        # Remove duplicates by converting to set, then back to list of lists
        return [list(t) for t in set(result)]

File: arrays/3sum.rb

# @param {Integer[]} nums
# @return {Integer[][]}
def three_sum(nums)
    result = []

    nums.each_with_index do |num, i|
            nums.each_with_index do |num, j|
                next if i == j
                nums.each_with_index do |num, k|
                    next if i == k || j == k
                        triplet = [nums[i], nums[j], nums[k]]
                        puts "i: #{i}"
                        puts "j: #{j}"
                        puts "k: #{k}"
                    
                    return [] if !triplet

                    result << triplet if triplet.sort!.sum == 0
                end
            end
    end

    result.uniq

end